### PR TITLE
Set closed_at inside update_all when we close tasks

### PR DESCRIPTION
**Pull request by pkarman**
In https://github.com/department-of-veterans-affairs/dsva-vacols/issues/107 we learned that some cancelled tasks do not have a `closed_at` value set.

This PR is a start of remedying that.

See also #13058

### CODE REVIEW

File: app/models/tasks/assign_hearing_disposition_task.rb [hunk 1]
```diff
@@ -110,7 +110,9 @@ def hold!
   private
 
   def update_children_status_after_closed
-    children.open.update_all(status: status)
+    update_args = { status: status }
+    update_args[:closed_at] = Time.zone.now if self.class.closed_statuses.include?(status)
+    children.open.update_all(update_args)
```
Comment (hschallhorn) [inline app/models/tasks/assign_hearing_disposition_task.rb]: This should always be true as `update_children_status_after_closed` is only called if `task_just_closed?` https://github.com/department-of-veterans-affairs/caseflow/blob/0f9e3874e346472cf8f7b0b77401666889b39f03/app/models/task.rb#L627

File: app/models/tasks/schedule_hearing_task.rb [hunk 2]
```diff
@@ -54,9 +54,9 @@ def create_change_hearing_disposition_task(instructions = nil)
     end
 
     # cancel my children, myself, and my hearing task ancestor
-    children.open.update_all(status: Constants.TASK_STATUSES.cancelled)
-    update!(status: Constants.TASK_STATUSES.cancelled)
-    ancestor_task_of_type(HearingTask)&.update!(status: Constants.TASK_STATUSES.cancelled)
+    children.open.update_all(status: Constants.TASK_STATUSES.cancelled, closed_at: Time.zone.now)
+    update!(status: Constants.TASK_STATUSES.cancelled, closed_at: Time.zone.now)
+    ancestor_task_of_type(HearingTask)&.update!(status: Constants.TASK_STATUSES.cancelled, closed_at: Time.zone.now)
```
Comment (hschallhorn) [inline app/models/tasks/schedule_hearing_task.rb]: Both the `update!`s here should be caught by the `set_timestamp` callback https://github.com/department-of-veterans-affairs/caseflow/blob/0f9e3874e346472cf8f7b0b77401666889b39f03/app/models/task.rb#L672

File: app/models/tasks/assign_hearing_disposition_task.rb (hunk 1 above)
Comment (pkarman) [reply app/models/tasks/assign_hearing_disposition_task.rb]: understood. Trying to be more explicit as a way of unwinding reliance on context/callbacks.

File: app/models/tasks/schedule_hearing_task.rb (hunk 2 above)
Comment (pkarman) [reply app/models/tasks/schedule_hearing_task.rb]: understood. Trying to be more explicit as a way of unwinding reliance on context/callbacks.

**Pull request merged**